### DISCUSSION TITLE
Speeding up the rendering of large shapefiles...

### DISCUSSION THREAD

**Discussion by keince** [pelicanmapping/osgearth#1862] [Support Q&A] [answered]
I have a large shapefile that defines country boundaries that I would like to speed up the rendering. When I enable this shapefile, I see a 12fps performance hit in the overall fps.

I tried generating the qix index file for the shapefile with the command:

**ogrinfo -sql "CREATE SPATIAL INDEX ON world" world.shp**

which generated the world.qix file. However, it didn't seem to make any difference in speeding things up.

My earth file has the following entry in it for the shapefile:

```
<OGRFeatures name="world-data">
     <url>C:/world.shp</url>`
     <filters>
          <convert type="line"/>
     </filters>
 </OGRFeatures>

 <FeatureModel name="Country boundaries" features="world-data" enabled="true" visible="true">
     <layout tile_size="500000" crop_features="true" paged="true">
         <level max_range="1e10"/>
     </layout>
     <styles
         <style type="text/css">
              world {
              stroke: #ffff00;
              stroke-width: 3px;
              stroke-tessellation-size: 1km;
              render-lighting: false;
              altitude-clamping: terrain-drape;
              }
         </style>
     </styles>
 </FeatureModel>
```

Is there something else I need to do to get this to work? If not, are there any other suggestions to speed up the rendering of large shapefiles?

Thx

Shayne

**Comment by plevy** [comment]
Any chance this is a publically available shapefile?

How many features? Is it many features in a small area?  

My first thought is use max_range="20000" and not "1e10"

**Reply by keince** [reply]
The shapefile is just the world.shp in the data folder. Not too large.

**Comment by gwaldron** [comment, marked as answer]
Consider going with a `FeatureImage` instead of a `FeatureModel` layer.

**Reply by keince** [reply]
Glenn, 

Changing over from FeatureModel to FeatureImage yielded a significant gain in performance. When zoomed out viewing the entire globe, FeatureModel was rendering at 40fps. Switching over to FeatureImage bumped the rate back up to 60fps.

Thanks for the suggestion!

**Reply by keince** [reply]
I've noticed that when the shapefile is "large" like 150Mb, the FeatureImage goes to an unacceptable crawl paging in higher detail. I'm hoping there's a preprocessing step I can take to speed up paging.
